File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    # 패턴을 선택하는 사람들의 답안을 미리 정의
    patterns = [
        [1, 2, 3, 4, 5], # 1번 수포자 패턴
        [2, 1, 2, 3, 2, 4, 2, 5],   # 2번 수포자 패턴
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]  # 3번 수포자 패턴
    ]
    
    scores = [0] * len(patterns) # 각 수포자의 점수를 저장할 리스트
    
    # 각 문제의 답안과 수포자들의 패턴을 비교하여 점수를 계산
    for idx, answer in enumerate(answers):
        for i, pattern in enumerate(patterns):
            if pattern[idx % len(pattern)] == answer:
                scores[i] += 1
    
    # 가장 높은 점수를 받은 수포자를 추출
    max_score = max(scores)
    
    answer = [i + 1 for i, score in enumerate(scores) if score == max_score]
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        ArrayList<Integer> answer = new ArrayList<>();
        int[] counts = new int[3];
        
        int[] pattern1 = {1, 2, 3, 4, 5};
        int[] pattern2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] pattern3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        for (int i = 0; i < answers.length; i ++){
            if(pattern1[i%pattern1.length] == answers[i]) {
                counts[0]++;
            }
            
            if(pattern2[i%pattern2.length] == answers[i]) {
                counts[1]++;
            }
            
            if(pattern3[i%pattern3.length] == answers[i]) {
                counts[2]++;
            }
        }
        
        int max = Arrays.stream(counts).max().getAsInt();
        
        for (int i = 0; i < counts.length; i++){
            if (counts[i] == max){
                answer.add(i+1);
            }
        }
        
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}